tests/plugins/clipboard/datatransfer.js

```diff
@@ -1252,5 +1252,106 @@ bender.test( {
 		var dataTransfer = new CKEDITOR.plugins.clipboard.dataTransfer();
 
 		arrayAssert.itemsAreSame( [], dataTransfer.getTypes() );
+	},
+
+	// (#4604)
+	'test asynchronous getTypes (in browsers with custom types support)': function() {
+		if ( !CKEDITOR.plugins.clipboard.isCustomDataTypesSupported ) {
+			return assert.ignore();
+		}
+
+		var nativeData = bender.tools.mockNativeDataTransfer(),
+			dataTransfer = new CKEDITOR.plugins.clipboard.dataTransfer( nativeData ),
+			expectedTypes = [ 'hublabubla', 'cke4/custom' ];
+
+		nativeData.types = expectedTypes;
+		dataTransfer.cacheData();
+
+		// When asynchronously accessing native data transfer it is unavailable. Here we simulate it by removing stored types.
+		nativeData.types = [];
+
+		arrayAssert.itemsAreSame( expectedTypes, dataTransfer.getTypes() );
+	},
+
+	// (#4604)
+	'test asynchronous getTypes (in browsers without custom types support)': function() {
```

Comment:
Could you rename all test to better cover those points:
1. what is tested
2. what are the conditions
3. what we are expecting

In this one we have
1. getTypes
2. asynchronous, browsers without custom types support
3. -- none --

So we are missing our expectations. It could be renamed to something like: `test asynchronous getTypes in browsers without custom types support returns original types`. It also helps to avoid such a sentence as `detecting correctly`. We have to read the code to find out what does that means. And if the code has some bugs - there is some confusion. Should we trust the test code or the logic code? With this test naming convention, we can express our `correctly` conception in plain words.

Please make the same process for the other tests.

It is much clearer to understand what is test doing without looking into code.